test/fast/Sourcing nvm.sh should not modify parameters of caller

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+set -- yes
```

Comment:
I forgot the word _not_ in my comment :/

`set` changes the arguments of the current script. `echo 'set -- A; echo $1' | . /dev/stdin` and `echo 'echo $1' | . /dev/stdin A` are equivalent.

The problem I am fixing only occurs if you do _not_ pass a parameter to `. ./nvm.sh`. If you do pass parameters `nvm.sh` gets it own set of parameters and it cannot modify the callers parameters.